pagefaultgames/pokerogue
# [Feature] Move touch controls configuration

## What are the changes?

I've added a setting that opens a configuration panel to move the touch controls around the screen. This setting is only visible when the client has a touch screen. This setting does not use option values like the other settings, but instead opens the configuration panel using the action button. Both landscape mode and portrait mode have their own control positions saved separately.


## Why am I doing these changes?



I implemented these changes to improve the user experience by offering greater flexibility in the placement of the touch controls. This way the game is more accessible and comfortable to play on mobile devices. 

## What did change?


Most changes are inside a new singleton class (MoveTouchControlsHandler). This class is responsible for maintaining the configuration panel for the touch controls. Aside from this the CSS styling and HTML structure of the touch controls had to be changed to allow for these configuration options. Finally a new setting is added that can be activated using the action button that opens up the configuration panel.

### Screenshots/Videos

https://github.com/pagefaultgames/pokerogue/assets/43152351/9e4f2937-5c8a-45bd-a071-52a86272a10a





## How to test the changes?




Open the branch on a mobile device or a simulated mobile device using browser dev tools. Then navigate to the new setting on the bottom. This setting only appears when the client has a touch screen. By using the action button on the setting, a new panel opens up where you can drag the touch controls around the screen to reconfigure their position.
